entity-service-impl/src/main/java/org/hypertrace/entity/metric/EntityMetricsReporter.java

```diff
@@ -0,0 +1,81 @@
+package org.hypertrace.entity.metric;
+
+import static org.hypertrace.core.documentstore.expression.operators.AggregationOperator.COUNT;
+import static org.hypertrace.core.documentstore.model.config.CustomMetricConfig.VALUE_KEY;
+
+import java.time.Duration;
+import java.util.List;
+import org.hypertrace.core.documentstore.Datastore;
+import org.hypertrace.core.documentstore.expression.impl.AggregateExpression;
+import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+import org.hypertrace.core.documentstore.expression.impl.LogicalExpression;
+import org.hypertrace.core.documentstore.expression.impl.RelationalExpression;
+import org.hypertrace.core.documentstore.expression.operators.RelationalOperator;
+import org.hypertrace.core.documentstore.model.config.CustomMetricConfig;
+import org.hypertrace.core.documentstore.query.Filter;
+import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.serviceframework.docstore.metrics.DocStoreCustomMetricReportingConfig;
+import org.hypertrace.core.serviceframework.docstore.metrics.DocStoreMetricsRegistry;
+import org.hypertrace.entity.v1.entitytype.EntityType;
+
+public class EntityMetricsReporter {
+
+  private static final String ALL_API_COUNT_METRIC_NAME = "all.api.entities.count";
+  private static final String RAW_ENTITIES_COLLECTION = "raw_entities";
+  private static final String API_DISCOVERY_STATE_ENTITY_PATH =
+      "attributes.api_discovery_state.value.string";
+  private static final String TENANT_ID_ENTITY_PATH = "tenantId";
+  private static final String ENTITY_TYPE_ENTITY_PATH = "entityType";
+  private static final String DISCOVERED = "DISCOVERED";
+  private static final String UNDER_DISCOVERY = "UNDER_DISCOVERY";
+  private final DocStoreMetricsRegistry metricsRegistry;
+
+  public EntityMetricsReporter(final Datastore datastore) {
+    metricsRegistry = new DocStoreMetricsRegistry(datastore).withCustomMetrics(apiCounterConfig);
+  }
+
+  public void monitor() {
+    metricsRegistry.monitor();
+  }
+
+  @SuppressWarnings("FieldCanBeLocal")
+  private final List<DocStoreCustomMetricReportingConfig> apiCounterConfig =
+      List.of(
+          DocStoreCustomMetricReportingConfig.builder()
+              .reportingInterval(Duration.ofHours(1))
+              .config(
+                  CustomMetricConfig.builder()
+                      .metricName(ALL_API_COUNT_METRIC_NAME)
+                      .collectionName(RAW_ENTITIES_COLLECTION)
+                      .query(
+                          Query.builder()
+                              .setFilter(getFilter())
+                              .addSelection(IdentifierExpression.of(TENANT_ID_ENTITY_PATH))
+                              .addSelection(
+                                  IdentifierExpression.of(ENTITY_TYPE_ENTITY_PATH), "entity_type")
```

Comment:
Can we either alias both tenant id and entity type or avoid aliasing both for consistency?
That way, all the label keys would be either in camelCase or in snake_case (but not a mix of both).

Reply:
On a separate note, given that we are already filtering by entity type API, we don't need this aggregation (since the label will always be 'API').